src/main/java/uk/gov/justice/probation/courtcasematcher/restclient/OffenderSearchRestClient.java

```diff
@@ -56,11 +57,11 @@ public OffenderSearchRestClient(@Qualifier("offenderSearchWebClient") WebClient
         this.webClient = webClient;
     }
 
-    public Mono<SearchResponse> search(Name name, LocalDate dateOfBirth){
+    public Mono<SearchResponse> search(String pnc, Name name, LocalDate dateOfBirth){
```

Comment:
I did attempt a refactor to change this to pass a single MatchRequest parameter in place of these three. It involved moving the validation and construction of the MatchRequest out of this class, I just created a static `from(pnc, Name, dateOfBirth)` method on MatchRequest. I ditched it in the end because moving the validation outside of this class had knock on effects and I didn't want to get too sidetracked.

It did have the nice side effect of making the MatchRequest creation and validation fully unit-testable though.

What do you think? Is this worth pursuing for the sake of a couple of hours more work?

Reply:
Happy either way. We are probably still a little light on the backlog so you could do it but I'm not overly concerned.